packages/terra-toggle-section-header/src/ToggleSectionHeader.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'terra-base/lib/baseStyles';
+import SectionHeader from 'terra-section-header';
+import Toggle from 'terra-toggle';
+
+const VALID_LEVELS = [1, 2, 3, 4, 5, 6];
+
+const propTypes = {
+  /**
+   * Content in the body of the toggle section header component that will be expanded or hidden.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Text to be displayed in the section header.
+   */
+  title: PropTypes.string.isRequired,
+  /**
+   * Sets the heading level. One of 1, 2, 3, 4, 5, or 6.
+   */
+  level: PropTypes.oneOf(VALID_LEVELS),
+  /**
+   * Callback function triggered when toggle-section-header is closed.
+   */
+  onClose: PropTypes.func,
+  /**
+   * Callback function triggered when toggle-section-header is opened.
+   */
+  onOpen: PropTypes.func,
+  /**
+   * Used to set props and HTML attributes on the underlying section-header.
+   */
+  sectionHeaderAttrs: PropTypes.object,
+  /**
+   * Sets the toggle-section-header to be animated when it is opened or closed.
+   */
+  isAnimated: PropTypes.bool,
+  /**
+   * Sets the toggle-section-header initial state to open.
+   */
+  isInitiallyOpen: PropTypes.bool,
+};
+
+const defaultProps = {
+  level: 2,
+  sectionHeaderAttrs: {},
+};
+
+class ToggleSectionHeader extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { isInitiallyOpen = false } = this.props;
+    this.state = { isOpen: isInitiallyOpen };
+
+    this.handleOnClick = this.handleOnClick.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isInitiallyOpen !== this.props.isInitiallyOpen) {
+      this.setState({ isOpen: nextProps.isInitiallyOpen });
+    }
+  }
+
+  handleOnClick(event) {
```

Comment:
Although onClick is not a first class prop, we will want to honor any custom onClick prop passed in. We should check for it and trigger it.

Reply:
This component is intended to serve similar functionality to `terra-toggle-button`. I would expect any `onClick` callback prop to be passed via `sectionHeaderAttrs`, as its the underlying functional component. Similarly to how [`terra-toggle-button`](https://github.com/cerner/terra-core/blob/toggle-section-header/packages/terra-toggle-button/src/ToggleButton.jsx#L85-L98) handles this.

Reply:
`onClick` would not be honors by passing via the `sectionHeaderAttrs` because we are providing an onClick ourselves, so it would need to be handed in this instance.

Reply:
Added wrapper to the function here d66c0cb234ac0cdb20b969ad6feeba59c1781871

Reply:
This isn't quite what I was thinking. Given you feel the onClick should be passed via the`sectionHeaderAttrs`, I was thinking this would be handling as such:
```diff
   handleOnClick(event) {
     event.preventDefault();

-     const { onOpen, onClose  } = this.props;
+     const { onOpen, onClose, sectionHeaderAttrs  } = this.props;
 
     if (onOpen && !this.state.isOpen) {
       onOpen();
     } else if (onClose && this.state.isOpen) {
       onClose();
     }
     
+    if (sectionHeaderAttrs.onClick) {
+        sectionHeaderAttrs.onClick();
+    }

     this.setState(prevState => ({
       isOpen: !prevState.isOpen,
     }));
   }
```
I would not expect to pass the customProps.onClick to the section header as there is a specific section header attributes prop

Reply:
As section-header accepts an `onClick` function, I would expect the responsibility of calling the `onClick` we provide here to be delegated to it, which is why I chose to wrap it. However, I think you're right in the fact that as i said before, I think the onClick should be provided via the `sectionHeaderAttrs`, not `customProps`, i'll make that update